contents/docs/integrate/_snippets/code-variants/ios-swiftui.mdx

```diff
@@ -0,0 +1,26 @@
+```swift
+import SwiftUI
+import PostHog
+
+@main
+struct YourGreatApp: App {
```

Comment:
There are more ways to do this, for example:

```suggestion
struct YourGreatApp: App {

  @UIApplicationDelegateAdaptor(AppDelegate.self) var appDelegate
  
  init() {}
```

and then:

```swift
class AppDelegate: NSObject, UIApplicationDelegate {
    func application(_: UIApplication, didFinishLaunchingWithOptions _: [UIApplication.LaunchOptionsKey: Any]? = nil) -> Bool {
        // init posthog
        return true
    }
}
```

Maybe add a comment that they can also use (or already use) the `UIApplicationDelegateAdaptor`, they can place the code inside the `application` method? or even another tab using the delegate?

@ioannisj might have other ideas.

Reply:
Oh I do like the thought of one more tab so we cover all the common cases:

- UIKit
- SwiftUI App init()
- UIApplicationDelegateAdaptor

Minimal mental friction for whichever setup you’re using.

Reply:
How about amending the comment in the above screenshot to read:

```swift
//Add PostHog to your app's initializer. If using UIApplicationDelegateAdaptor, see the UIKit tab.
```

Would that get us there?

Reply:
Sold! 👨🏻‍⚖️

Thank you both!